auth/service.go

```diff
@@ -170,7 +191,58 @@ func (svc service) Identify(ctx context.Context, token string) (Identity, error)
 	}
 }
 
-func (svc service) Authorize(ctx context.Context, token, sub, obj, act string) (bool, error) {
+func (svc service) Authorize(ctx context.Context, check bool, sub, obj, act string) (bool, error) {
+	return svc.authorize(ctx, check, sub, obj, act)
+}
+
+func (svc service) authorize(ctx context.Context, check bool, sub, obj, act string) (bool, error) {
+	if check {
+		conn, err := grpc.Dial(fmt.Sprintf("%s:%s", ketoContainerName, svc.ketoConfig.ReadPort), grpc.WithInsecure())
+		if err != nil {
+			return false, err
+		}
+		client := acl.NewCheckServiceClient(conn)
+		res, err := client.Check(context.Background(), &acl.CheckRequest{
+			Namespace: ketoNamespace,
+			Object:    obj,
+			Relation:  act,
+			Subject: &acl.Subject{Ref: &acl.Subject_Id{
+				Id: sub,
+			}},
+		})
+		if err != nil {
+			return false, err
+		}
+
+		if res.Allowed {
```

Comment:
Good job Burak!
what about `return res.Allowed, nil`

Reply:
Hi @tzzed, thanks. 
Your suggestion seems more appropriate than my solution. I guess it is better to return with `res.Allowed`. I will update the PR.